Sources/RobotKit/Actions/Notification.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  Notification.swift
+//  RobotKit
+//
+//  Created by Ahmed Shendy on 29/08/2022.
+//
+
+import Foundation
+import UserNotifications
+
+/// Robot control of notification center and its delegate
+public actor RobotNotification: SelfTasking {
+    
+    private class DefaultNotificationCenterDelegate: NSObject, UNUserNotificationCenterDelegate { }
+    
+    private var current: UNUserNotificationCenter = UNUserNotificationCenter.current()
+    public var delegate: UNUserNotificationCenterDelegate = DefaultNotificationCenterDelegate() {
+        didSet {
+            current.delegate = delegate
+        }
+    }
+    
+    public var isAuthorized: Bool {
+        let settings = await current.notificationSettings()
+        
+        return settings.authorizationStatus == .authorized
+    }
```

Comment:
I think this might fix the async error...

```swift
@MainActor
public var isAuthorized: Bool {
    let settings = current.notificationSettings()

    return settings.authorizationStatus == .authorized
)
```